src/logic/safe/store/actions/processTransaction.ts

```diff
@@ -63,143 +38,53 @@ interface ProcessTransactionArgs {
-      safeInstance,
+    const txProps = {
+      navigateToTransactionsTab: false,
+      notifiedTransaction: props.notifiedTransaction,
+      operation: tx.operation,
+      origin: tx.origin,
+      safeAddress: props.safeAddress,
       to: tx.to,
+      txData: tx.data ?? EMPTY_DATA,
+      txNonce: '',
       valueInWei: tx.value,
-      data: tx.data ?? EMPTY_DATA,
-      operation: tx.operation,
-      nonce: tx.nonce,
       safeTxGas: tx.safeTxGas,
-      baseGas: tx.baseGas,
-      gasPrice: tx.gasPrice || '0',
-      gasToken: tx.gasToken,
-      refundReceiver: tx.refundReceiver,
-      sender: from,
-      sigs,
+      ethParameters: props.ethParameters,
+      delayExecution: !props.approveAndExecute,
     }
 
+    // Populate instance vars
     try {
-      if (checkIfOffChainSignatureIsPossible(isExecution, smartContractWallet, safeVersion)) {
-        const signature = await tryOffChainSigning(
-          tx.safeTxHash,
-          { ...txArgs, safeAddress },
-          hardwareWallet,
-          safeVersion,
-        )
-
-        if (signature) {
-          dispatch(closeSnackbarAction({ key: beforeExecutionKey }))
-
-          await saveTxToHistory({ ...txArgs, signature })
-
-          dispatch(fetchTransactions(chainId, safeAddress))
-          return
-        }
-      }
-
-      transaction = isExecution ? getExecutionTransaction(txArgs) : getApprovalTransaction(safeInstance, tx.safeTxHash)
-
-      const sendParams: PayableTx = {
-        from,
-        value: 0,
-        gas: ethParameters?.ethGasLimit,
-        [getGasParam()]: ethParameters?.ethGasPriceInGWei,
-        nonce: ethParameters?.ethNonce,
-      }
-
-      await transaction
-        .send(sendParams)
-        .once('transactionHash', async (hash: string) => {
-          txHash = hash
-          dispatch(closeSnackbarAction({ key: beforeExecutionKey }))
-
-          if (isExecution) {
-            dispatch(updateTransactionStatus({ safeTxHash: tx.safeTxHash, status: LocalTransactionStatus.PENDING }))
-          }
-
-          try {
-            await saveTxToHistory({ ...txArgs })
-
-            // store the pending transaction's nonce
-            isExecution && aboutToExecuteTx.setNonce(txArgs.nonce)
-
-            dispatch(fetchTransactions(chainId, safeAddress))
-          } catch (e) {
-            logError(Errors._804, e.message)
-          }
-        })
-        .then(async (receipt) => {
-          dispatch(fetchTransactions(chainId, safeAddress))
-
-          if (isExecution) {
-            dispatch(fetchSafe(safeAddress))
-          }
-
-          return receipt.transactionHash
-        })
+      await sender.prepare(dispatch, state, txProps)
     } catch (err) {
-      logError(Errors._804, err.message)
-
-      dispatch(closeSnackbarAction({ key: beforeExecutionKey }))
-
-      if (isExecution) {
-        dispatch(updateTransactionStatus({ safeTxHash: tx.safeTxHash, status: LocalTransactionStatus.PENDING_FAILED }))
-      }
+      return
```

Comment:
same here with error hiding

Reply:
I'll add some logging here. Thanks for noticing it!